### PR TITLE
search: rename SearchSymbolResult -> SymbolMatch

### PR DESCRIPTION
**Pull request by rvantonder**
Stacked on #18844.

As per description.

### CODE REVIEW

File: internal/search/result/symbol.go [hunk 1]
```diff
@@ -22,13 +22,13 @@ type Symbol struct {
...
 type Symbols = []Symbol
 
-// SearchSymbolResult is a result from symbol search.
-type SearchSymbolResult struct {
+// SymbolMatch is a symbol search result decorated with extra metadata in the frontend.
+type SymbolMatch struct {
 	Symbol  Symbol
 	BaseURI *gituri.URI
 	Lang    string
 }
```
Comment (rvantonder) [inline internal/search/result/symbol.go:30]: This is the key change, everything else just propagates the change.

I was toying with the idea of calling this `DecoratedSymbol`, but it's a bit more verbose. Either way `result.SymbolMatch` is a whole lot better than `result.SearchSymbolResult`.

Random thing I didn't investigate: why is there a `Lang` member in this type and also a `Language` member inside `Symbol`? Are these duplicated?

Comment (camdencheek) [reply internal/search/result/symbol.go:30]: RE: `Lang`, I remember looking at that when I refactored this out of the resolver, and I seem to remember seeing some inconsistencies there that I couldn't change without changing behavior, so I punted at the time. As an example, [here](https://sourcegraph.com/github.com/sourcegraph/sourcegraph@d75922801cf252b31550d7f973c86b88b77a12cf/-/blob/cmd/frontend/graphqlbackend/symbols.go#L154-L169), we set lang from the file, but don't populate the symbol's `Language`.

**Review by camdencheek — APPROVED** [review]
I've been looking forward to this one 💯

File: internal/search/result/symbol.go (hunk 1 above)
Comment (rvantonder) [reply internal/search/result/symbol.go:30]: ohh got it. so it's more like file metadata or something.

**Pull request merged**